### PR TITLE
CSLP-1740: Merge Feature appframework to develop

### PR DESCRIPTION
**Pull request by vivekr-splunk**
- merging appframework to develop
- changes to configmap reconcilation, based on ticket https://splunk.atlassian.net/browse/CSPL-1782

### CODE REVIEW

File: pkg/splunk/enterprise/util.go [hunk 1]
```diff
@@ -31,17 +36,97 @@ import (
...
 
 // kubernetes logger used by splunk.enterprise package
 //var log = logf.Log.WithName("splunk.enterprise")
 
+var operatorResourceTracker *globalResourceTracker = nil
+
+// initialize operator level context
+func init() {
+	fmt.Printf("init is called here\n")
+	defer fmt.Printf("init ends here\n")
```
Comment (gaurav-splunk) [inline pkg/splunk/enterprise/util.go:59]: were these intentional ?

**Review by gaurav-splunk — APPROVED** [review]

File: pkg/splunk/client/enterprise.go [hunk 2]
```diff
@@ -27,12 +27,11 @@ import (
 	"time"
 
 	splcommon "github.com/splunk/splunk-operator/pkg/splunk/common"
-
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	//logf "sigs.k8s.io/controller-runtime/pkg/log"
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply pkg/splunk/client/enterprise.go:30]: Fixed

File: pkg/splunk/client/enterprise.go [hunk 3]
```diff
@@ -27,12 +27,11 @@ import (
...
 )
 
 // kubernetes logger used by splunk.enterprise package
-var log = logf.Log.WithName("splunk.client")
+//var log = logf.Log.WithName("splunk.client")
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply pkg/splunk/client/enterprise.go:34]: Fixed

File: pkg/splunk/common/util.go [hunk 4]
```diff
@@ -29,7 +29,7 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	//logf "sigs.k8s.io/controller-runtime/pkg/log"
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply pkg/splunk/common/util.go:32]: Fixed

File: pkg/splunk/enterprise/util.go (hunk 1 above)
Comment (vivekr-splunk) [reply pkg/splunk/enterprise/util.go:59]: yes was used in debugging, removed for now

**Review by jryb — APPROVED** [review]

File: controllers/searchheadcluster_controller_test.go [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+	//ctrl "sigs.k8s.io/controller-runtime"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	//"k8s.io/client-go/kubernetes/scheme"
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply controllers/searchheadcluster_controller_test.go:24]: @kashok-splunk can you address this in your PR

File: controllers/searchheadcluster_controller_test.go [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+	//"reflect"
+	"time"
+
+	enterpriseApi "github.com/splunk/splunk-operator/api/v3"
+	"github.com/splunk/splunk-operator/controllers/testutils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	//ctrl "sigs.k8s.io/controller-runtime"
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply controllers/searchheadcluster_controller_test.go:16]: @kashok-splunk can you address this in your PR

File: controllers/searchheadcluster_controller_test.go [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+			request := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      "test",
+					Namespace: namespace,
+				},
+			}
+			// econcile for the first time err is resource not found
+			_, err := instance.Reconcile(ctx, request)
+			Expect(err).ToNot(HaveOccurred())
+			// create resource first adn then reconcile for the first time
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply controllers/searchheadcluster_controller_test.go:108]: @kashok-splunk can you address this in your PR

File: controllers/searchheadcluster_controller_test.go [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+				Client: c,
+				Scheme: scheme.Scheme,
+			}
+			request := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      "test",
+					Namespace: namespace,
+				},
+			}
+			// econcile for the first time err is resource not found
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply controllers/searchheadcluster_controller_test.go:105]: @kashok-splunk can you address this in your PR

File: controllers/suite_test.go [hunk 9]
```diff
@@ -82,14 +103,156 @@ var _ = BeforeSuite(func() {
...
+		Expect(err).NotTo(HaveOccurred())
+	}
+	if err := (&StandaloneReconciler{
+		Client: k8sManager.GetClient(),
+		Scheme: k8sManager.GetScheme(),
+	}).SetupWithManager(k8sManager); err != nil {
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	/*
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply controllers/suite_test.go:148]: @kashok-splunk can you address this in your PR

File: controllers/suite_test.go [hunk 10]
```diff
@@ -82,14 +103,156 @@ var _ = BeforeSuite(func() {
...
 	Expect(k8sClient).NotTo(BeNil())
 
 }, 60)
 
 var _ = AfterSuite(func() {
 	By("tearing down the test environment")
-	err := testEnv.Stop()
-	Expect(err).NotTo(HaveOccurred())
+	testEnv.Stop()
+	//Expect(err).NotTo(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply controllers/suite_test.go:170]: @kashok-splunk can you address this in your PR

File: bundle.Dockerfile [hunk 11]
```diff
@@ -6,7 +6,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=splunk-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.15.0
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.18.0
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply bundle.Dockerfile:9]: @kashok-splunk can you address this in your PR

**Review by kashok-splunk — APPROVED** [review]

File: config/manager/manager.yaml [hunk 12]
```diff
@@ -60,9 +61,22 @@ spec:
...
+          - mountPath: /opt/splunk/appframework/
+            name: app-staging
       serviceAccountName: controller-manager
       volumes:
       - configMap:
           name: config
         name: splunk-operator-config
+      - name:  app-staging
+        persistentVolumeClaim:
+          claimName: tmp-app-download
```
Comment (sgontla) [inline config/manager/manager.yaml:81]: Nit: Can we remove the `tmp` word, considering it is PVC?

File: controllers/monitoringconsole_controller_test.go [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+	//"reflect"
+	"time"
+
+	enterpriseApi "github.com/splunk/splunk-operator/api/v3"
+	"github.com/splunk/splunk-operator/controllers/testutils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	//ctrl "sigs.k8s.io/controller-runtime"
```
Comment (sgontla) [inline controllers/monitoringconsole_controller_test.go:16]: nit: commented code removal, same for other places.

**Review by sgontla — APPROVED** [review]

File: config/manager/manager.yaml (hunk 12 above)
Comment (vivekr-splunk) [reply config/manager/manager.yaml:81]: fixed

**Pull request merged**